### ISSUE TITLE
creating new instance of the DefaultKubernetesClient is hanging 

### ISSUE THREAD

**Issue by umutkocasarac** [fabric8io/kubernetes-client#2458]
Creating new instance of the DefaultKubernetesClient is hanging and never responding. I could see the following messages  on the logs

```
16:36:38.471 [main] DEBUG io.fabric8.kubernetes.client.Config - Trying to configure client from Kubernetes config...
16:36:38.471 [main] DEBUG io.fabric8.kubernetes.client.Config - Found for Kubernetes config at: [/Users/----/.kube/config].
```

The reason for that is kubernetes client using the config file and trying to get token from the AWS with the aws cli. However there is an issue with the following code

```java
Process p = pb.start();
      if (p.waitFor() != 0) {
        LOGGER.warn(IOHelpers.readFully(p.getErrorStream()));
      }
      ExecCredential ec = Serialization.unmarshal(p.getInputStream(), ExecCredential.class);
```

ProcessBuilder should read the output of the command before waiting the result of the execution. It seems like an easy fix, I could provide a pull request if you want

**Comment by rohanKanojia** [comment]
@umutkocasarac: Please go ahead and create a PR to fix this.

**Comment by rohanKanojia** [comment]
Are you facing this issue after upgrade to v4.11.0 or are you using client for the first time?

**Comment by rohanKanojia** [comment]
I think you should be facing this issue on older versions too since we haven't changed this code in recent releases.

**Comment by umutkocasarac** [comment]
I am getting this error on both v4.10.3 and v4.11.0 versions

**Comment by rohanKanojia** [comment]
pheww, It's not a regression :-) . Actually, since we released 4.11.0 people have reported a few regressions. I was a bit worried if this is another regression.

**Comment by umutkocasarac** [comment]
hey @rohanKanojia I am getting permission error while pushing my changes. Do I need to do anything before pushing my changes to branch?

kubernetes-client % git push origin iss_2458
ERROR: Permission to fabric8io/kubernetes-client.git denied to umutkocasarac.
fatal: Could not read from remote repository.

**Comment by rohanKanojia** [comment]
oh, you need to fork the repository and then push to your own fork.

1. Fork Fabric8 Github repository to your account
2. Add your fork remote `git remote add my-fork https://github.com/umutkocasarac/kubernetes-client.git`
3. Push to your fork `git push my-fork iss_2458`
4. Make PR against fabric8 master from Github UI :rocket:

**Comment by manusa** [comment]
We assume that this issue was fixed with #2487. Please open another issue in case you're still facing problems.
